### PR TITLE
[ci] Update golangci

### PR DESCRIPTION
**Pull request by patrick-ogrady**

### CODE REVIEW

**Review by richardpringle — COMMENTED** [review]
I'm fine with this, but @dboehm-avalabs might want his stuff merged before this monster

**Comment by patrick-ogrady** [comment]
> I'm fine with this, but @dboehm-avalabs might want his stuff merged before this monster

@dboehm-avalabs is doing a deeper rewrite that just pertains to the `x/programs` package (not yet pushed public). This cleans up a number of issues all over the place. His rebase will be minimal.

File: gossiper/proposer.go [hunk 1]
```diff
@@ -115,7 +116,7 @@ func (g *Proposer) Force(ctx context.Context) error {
 	mempoolErr := g.vm.Mempool().Top(
 		ctx,
 		g.vm.GetTargetGossipDuration(),
-		func(ictx context.Context, next *chain.Transaction) (cont bool, rest bool, err error) {
```
Comment (patrick-ogrady) [inline gossiper/proposer.go:118]: ❤️ that new linter

File: vm/vm.go [hunk 2]
```diff
@@ -42,9 +36,15 @@ import (
 	"github.com/ava-labs/hypersdk/network"
 	"github.com/ava-labs/hypersdk/rpc"
 	"github.com/ava-labs/hypersdk/state"
-	htrace "github.com/ava-labs/hypersdk/trace"
-	hutils "github.com/ava-labs/hypersdk/utils"
+	"github.com/ava-labs/hypersdk/trace"
+	"github.com/ava-labs/hypersdk/utils"
 	"github.com/ava-labs/hypersdk/workers"
+
+	avametrics "github.com/ava-labs/avalanchego/api/metrics"
```
Comment (patrick-ogrady) [inline vm/vm.go:43]: cool

**Pull request merged**